food-truck.eve.md

```diff
@@ -2,91 +2,105 @@
-    cost:12]
-
-   [#menu name:"Veggie Burger"
-    image:"https://pixabay.com/en/eat-burger-onion-avocado-veggi-1352880/"
-    description:"Our totally vegan black bean and portabella mushroom burger, topped with avocado, grilled onion, vegan cheese, and soy mayo, all on a gluten-free bun"
-    cost:12
-    #gluten-free
-    #vegetarian]
-
-   [#menu name:"Chicken Salad Sandwich"
-    image:"https://pixabay.com/en/bread-breakfast-brown-business-1239276/"
-    description:"Grandma’s chicken salad recipe with grapes and walnuts, served on wholesome 12 grain bread"
-    cost:10]
 
-   [#menu
-    name:"Charbroiled Chicken Wings"
-    image:"https://pixabay.com/en/barbecue-bbq-charcoal-chicken-88340/"
-    description:"Brined for 24 hours and coated with our secret spice rub, then grilled until then skin is crispy and the meat is juicy"
-    cost:10
-    #gluten-free
-    #spicy]
+Draw the remove-from-cart button for each menu item if this item is in the cart
+```
+search
+   [#app page:"homepage" order]
+   order-item = [#order-item order menu-item count]
+   not (count = 0)
+search @browser
+   d = [#div menu-item #container]
+bind @browser
+   d.children += [#div order-item #remove text:"remove from order" style:[border:"2px solid black" width:80]]
+```
 
+Add an item to the order after a click. in the original design this was a swipe right
+```
+search @browser @event @session
+   [#app page:"homepage" order]
+   [#click element:[#add menu-item]]
+   count = if [#order-item order menu-item count:c] then c + 1 else 1
+commit
+   order-item = [#order-item order menu-item]
+   order-item.count := count
+```
 
-   [#menu name:"French Fries"
-    image:"https://pixabay.com/en/french-fries-salt-food-923687/"
-    description:"Hand-cut Kennebec fries with Cajun seasoning"
-    cost:5
-    #gluten-free
-    #vegetarian]
+Handle the remove item from cart button
+in the original design this was swipe left, but we dont have swipe machinery yet
+```
+search @browser @event @session
+   [#app page:"homepage" order]
+   [#click element:[#remove order-item]]
+commit
+   order-item.count := order-item.count - 1
+```
 
 
-   [#menu
-    name:"Garlic Parmesan Mac & Cheese"
-    image:"https://pixabay.com/en/mac-and-cheese-pasta-food-cheese-521447/"
-    description:"Penne pasta smothered with aged cheddar, fresh garlic, and parsley"
-    cost:10
-    #vegetarian]
+display the shopping cart with count on the menu page.
+this currently lands at the bottom of the screen and should instead be the upper right
+      - Default is showing no number next to icon [TODO - always draw the cart]
+      - As items are added, count the items and display the count next to icon
+      - Clicking the icon takes you to the checkout page
+```
+search
+   [#app page:"homepage" order]
+   [#order-item order menu-item count]
+   item-count = sum[value: count per:order given:menu-item]
```

Comment:
if we wrap this and the `#order-item` record above in an `if`, we can have this always draw the cart and have the count be the empty string if there aren't any items.